### ISSUE TITLE
Include other values file from a values file

### ISSUE THREAD

**Issue by nodesocket** [helm/helm#7518]
Is it possible to include another values file in a values file passed via `helm install --name redis --values values.yaml ./`? I tried:

```
# values.yaml
default:
  foo: bar

redis:
  {{ .Files.get "redis.yaml" }}
```

```
# redis.yaml
# lots of values here
```

But this breaks with:

```
Error: failed to parse values.yaml: error converting YAML to JSON: yaml: invalid map key: map[interface {}]interface {}{".Files.get \"redis.yaml\"":interface {}(nil)}
```

**Comment by mattfarina** [comment]
It is not possible to use template functions within a `values.yaml` file. But, you can pass more than one file to the `helm` command. For example,

```
$ helm install --name redis --values foo.yaml --values bar.yaml
```

Helm will merge the values. Does this work?

**Comment by nodesocket** [comment]
@mattfarina awesome that worked.

FYI needed this because I am overriding values from  public chart using `requirements.yaml`.

```
dependencies:
  - name: redis
    version: 9.5.2
    repository: https://kubernetes-charts.storage.googleapis.com
```

```
helm dependencies update
```

```
helm install --name redis --values values/redis.yaml --values values/my-custom-values.yaml ./
```

**Comment by MurzNN** [comment]
If you need to include some file as value of variable, as replacement of integrating `app-config.ini` into `values.yaml` file manually like this:
```yaml
app:
  configContents: |-
    [group]
    value1=2
```
you can use this workaround:
```
helm install --set-file app.configContents=config-files/redis-config.ini
```
But this is unfortunately not working if you need to include other file as yaml structure ;(

**Comment by MurzNN** [comment]
Also I'd tried a way using Go Templates, something like this in `values.yaml` file:
```yaml
app:
  {{- .Files.Get "values-app.yaml" | nindent 2 }}
```
But it isn't work too :(

**Comment by norru** [comment]
Can this be made into a feature request? Implement `{{ include }}` in values files.

**Comment by jinleileiking** [comment]
merge not work

**Comment by zhaogaolong** [comment]
My requirement is to deploy the same chart on different clusters using different values files. In order to reduce the complexity of the entry values file, I use the method of include other values files to reduce the complexity.


why values.yaml  not support include other values yaml ? 
for example:

```yaml
# cluster-01-values.yaml
include:
- "cluster-01-app1-values.yaml"
- "cluster-01-app2-values.yaml"
```
like: https://docs.gitlab.com/ee/ci/yaml/#includelocal

**Comment by nikhilsbhat** [comment]
I ran into a similar problem managing configurations owned by multiple teams.

I ended up building **yamll**, which lets you compose YAML from local files, Git repositories, OCI registries, or HTTP sources into a single generated YAML file before it's consumed. It also supports dependency locking and tracing where values came from.

It doesn't require changes to the consuming application, so it might be a useful approach for this use case.

https://github.com/nikhilsbhat/yamll
